src/usr/local/www/services_dhcp.php

```diff
@@ -400,14 +400,14 @@ function validate_partial_mac_list($maclist) {
 
 		if (!$input_errors) {
 			/* make sure the range lies within the current subnet */
-			$subnet_start = ip2ulong(long2ip32(ip2long($ifcfgip) & gen_subnet_mask_long($ifcfgsn)));
-			$subnet_end = ip2ulong(long2ip32(ip2long($ifcfgip) | (~gen_subnet_mask_long($ifcfgsn))));
+			$subnet_start = gen_subnetv4($ifcfgip, $ifcfgsn);
+			$subnet_end = gen_subnetv4_max($ifcfgip, $ifcfgsn);
 
-			if (ip2ulong($_POST['range_from']) > ip2ulong($_POST['range_to'])) {
+			if (ip_greater_than($_POST['range_from'], $_POST['range_to'])) {
 				$input_errors[] = gettext("The range is invalid (first element higher than second element).");
 			}
 
-			if (ip2ulong($_POST['range_from']) < $subnet_start || ip2ulong($_POST['range_to']) > $subnet_end) {
+			if (!is_inrange_v4($_POST['range_from'], $subnet_start, $subnet_end)) {
```

Comment:
You should check $_POST['range_to'] to make sure it's inside the current subnet, no?